include/swift/AST/DiagnosticConsumer.h

```diff
@@ -163,7 +172,10 @@ class FileSpecificDiagnosticConsumer : public DiagnosticConsumer {
   ///
   /// If None, Note diagnostics are sent to every consumer.
   /// If null, diagnostics are suppressed.
-  Optional<DiagnosticConsumer *> ConsumerForSubsequentNotes = None;
+  Optional<ConsumerSpecificInformation *>
+      ConsumerSpecificInfoForSubsequentNotes = None;
+
+  bool WasAnErrorSuppressed = false;
```

Comment:
This shouldn't be necessary.

Reply:
Removed, but I did had a flag `hasAnErrorBeenConsumed`.